labman/db/tests/integration_tests.py

```diff
@@ -0,0 +1,375 @@
+    # gDNA extraction
+    ext_process, gdna_plate = create_gdna_extraction_process(
+        user, sample_plate)
+    # gDNA compression
+    comp_process, compressed_plate = create_compression_process(
+        user, [gdna_plate])
+    # gDNA compressed quantification
+    gdna_comp_quant_process = create_quantification_process(
+        user, compressed_plate)
+    # Normalization process
+    norm_process, norm_plate = create_normalization_process(
+        user, gdna_comp_quant_process)
+    # Library prep shotgun
+    shotgun_process, shotgun_plate = create_shotgun_process(user, norm_plate)
+    # Quantify library plate
+    shotgun_quant_process = create_quantification_process(user, shotgun_plate)
+    # Pooling process
+    pool_process = create_plate_pool_process(
+        user, shotgun_quant_process, shotgun_plate,
+        {'function': 'equal', 'parameters': {'total_vol': 60, 'size': 500}})
+    # Sequencing process
+    seq_process = create_sequencing_process(user, [pool_process.pool])
+    return seq_process
+
+
+@click.command()
+def integration_tests():
+    samples = get_samples()
+    user = User('test@foo.bar')
+    amplicon_seq_process = amplicon_workflow(user, samples)
+    shotgun_seq_process = shotgun_workflow(user, samples)
+
+    obs = amplicon_seq_process.generate_sample_sheet()
+    res = re.match(EXP_AMPLICON_SAMPLE_SHEET, obs)
+    if res is None:
+        raise ValueError(
+            'Amplicon sample sheet does not match expected regex:\n%s' % obs)
+
+    obs = shotgun_seq_process.generate_sample_sheet()
+    res = re.match(EXP_SHOTGUN_SAMPLE_SHEET, obs)
+    if res is None:
+        raise ValueError(
+            'Shotgun sample sheet does not match expected regex:\n%s' % obs)
+
+
+EXP_AMPLICON_SAMPLE_SHEET = r"""# PI,Admin,admin@foo.bar
+# Contact,Demo,Dude
+# Contact emails,demo@microbio.me,test@foo.bar
+\[Header\]
+IEMFileVersion,4
+Investigator Name,Admin
+Experiment Name,Run experiment \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{6}
+Date,\d{4}-\d{2}-\d{2}
+Workflow,GenerateFASTQ
+Application,FASTQ Only
+Assay,Amplicon
+Description,
+Chemistry,Default
+
+\[Reads\]
+151
+151
+
+\[Settings\]
+ReverseComplement,0
+
+\[Data\]
+Sample_ID,Sample_Name,Sample_Plate,Sample_Well,I7_Index_ID,index,Sample_Project,Description,,
+New_sequencing_run_\d{4}-\d{2}-\d{2}_\d{2}_\d{2}_\d{2}_\d{6},,,,,NNNNNNNNNNNN,,,,,"""  # noqa
+
+
+EXP_SHOTGUN_SAMPLE_SHEET = r"""# PI,Admin,admin@foo.bar
+# Contact,Demo,Dude
+# Contact emails,demo@microbio.me,test@foo.bar
+\[Header\]
+IEMFileVersion,4
+Investigator Name,Admin
+Experiment Name,Run experiment \d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{6}
+Date,\d{4}-\d{2}-\d{2}
+Workflow,GenerateFASTQ
+Application,FASTQ Only
+Assay,Metagenomics
+Description,
+Chemistry,Default
+
+\[Reads\]
+151
+151
+
+\[Settings\]
+ReverseComplement,0
+
+\[Data\]
+Lane,Sample_ID,Sample_Name,Sample_Plate,Sample_Well,I7_Index_ID,index,I5_Index_ID,index2,Sample_Project,Description
```

Comment:
This looks ok, and should work as long as the db is fresh -- otherwise, possible that the sample:index associations will not match

Reply:
That's correct  - and expected. If we want to ensure associations and results this is the only way of doing it.

Reply:
Sweet!